server/src/main/java/io/crate/execution/dml/upsert/UpdateToInsert.java

```diff
@@ -161,15 +161,17 @@ public UpdateToInsert(NodeContext nodeCtx,
                 this.columns.add(insertColumn);
             }
         }
-        for (var column : table.columns()) {
+        for (var ref : table.columns()) {
             // The Indexer later on injects the generated column values
             // We only include them here if they are provided in the `updateColumns` to validate
             // that users provided the right value (otherwise they'd get ignored and we'd generate them later)
-            if (column instanceof GeneratedReference && !Arrays.asList(updateColumns).contains(column.column().fqn())) {
+            if (ref instanceof GeneratedReference
+                    && !Arrays.asList(updateColumns).contains(ref.column().fqn())
```

Comment:
Question (because I faced that several times) - does `updateColumns` come from SQL statement? Isn't it specified int SQL form, ie we should use .contains(column.column().**sqlFqn**())?

Reply:
They're set in:

https://github.com/crate/crate/blob/dfae1614597eaa782cdb9e3a994f172c1ca1ecad/server/src/main/java/io/crate/expression/symbol/Assignments.java#L71